### ISSUE TITLE
rogue relations in MP

### ISSUE THREAD

**Issue by balhoff** [mgijax/mammalian-phenotype-ontology#2964]
MP has a few relations with rogue identifiers, I assume left over from past OBO format conversions. Examples:
 - 'occurs in': http://purl.obolibrary.org/obo/mp/mp-equivalent-axioms-subq#occurs_in
- 'preceded_by': http://purl.obolibrary.org/obo/mp/mp-equivalent-axioms-subq#preceded_by

and there are a few others. This will impact reasoning and application of phenotype patterns. These are easy to spot in the ["base" release of the ontology](http://purl.obolibrary.org/obo/mp/mp-base.owl).

cc @matentzn

**Comment by sbello** [comment]
Can I assume that any relation which does not display as an ID when I switch the view to IRI is a 'rogue' relation? If this is correct then we have:
after (4 definitions)
during (14)
exists_during (13)
has_central_participant (42)
has_quality (9)
occurs_in (546)
preceded_by (1)
results_from (6)

 It seems likely that occurs_in can be replaced with inheres_in.

**Comment by balhoff** [comment]
> Can I assume that any relation which does not display as an ID when I switch the view to IRI is a 'rogue' relation

That sounds right. :-)

Is occurs_in not used the same as the ['occurs in' from RO](
http://purl.obolibrary.org/obo/BFO_0000066)?

**Comment by sbello** [comment]
Hopefully it is but I'm not sure. I'll need to check how it is used. The MP also has occurs_in with the BFO ID but of course that is only used in 1 EQ statement

**Comment by sbello** [comment]
Some appear to be in EQ involving pathological entities (tumor, sclerosis, hemorrhage). I do not think these should use occurs_in from RO (MP:0009222, MP:0003991)
Others are for GO processes (MP:0013542, MP:0001872) where I think we do want to use occurs_in

**Comment by sbello** [comment]
Another problematic subset are those that use GO cellular component and some tissue. See for example abnormal motile primary cilium morphology (MP:0004131). We can't use occurs in and there is already an inheres in relationship for the cellular component.

**Comment by sbello** [comment]
topology in specific location - abnormal intrathoracic topology of vagus nerve
'has part' some 
    (position
     and ('inheres in' some 
        ('vagus nerve'
         and ('occurs in' some 'thoracic cavity')))
     and ('has modifier' some abnormal))
Need to determine the correct relation to replace 'occurs in' with - should it be 'part of'?

**Comment by sbello** [comment]
@cmungall I've been running into a weird bug in Protege (5.1.0) where when I 
open an EQ in the IRI view, 
then close the EQ and switch view to label, 
then reopen the same EQ with the text labels showing
then close the EQ and switch back to IRI view many of the IRIs in the EQ have been corrupted
Any idea what is going on?

**Comment by matentzn** [comment]
Hmm something other than the switch from PATO quality to BFO quality? That is serious.. can you give an exampl?

**Comment by sbello** [comment]
Yes, the has part at the top of the relation started as BFO_0000051 after the bug it displays as 'has part' when in the IRI view and the EQ is broken until I go back a replace it with the ID again. Other cases turned PATO terms into text plus some other PATO ID. I'll see if I can trigger it again and take a screen shot of the results

**Comment by matentzn** [comment]
Exact Version of protege as well. I will need to report this! Don’t wanna know how often this happens without anyone noticing it!

**Comment by sbello** [comment]
Before
![beforeeq](https://user-images.githubusercontent.com/10979793/53986515-61df6280-40ec-11e9-8f27-6170d9c85dff.PNG)
fixed the 'occurs in' to be the BFO ID, 
switch to label
open EQ click OK
switch to IRI and now the EQ is
![aftereq](https://user-images.githubusercontent.com/10979793/53986517-6441bc80-40ec-11e9-9eeb-a1e03ffd7773.PNG)

**Comment by sbello** [comment]
Version is 5.1.0
I have the ELK 0.4.3 reasoner running (in case that makes a difference)
I'm on a PC running Windows 7

**Comment by matentzn** [comment]
It’s time we start our face to face sessions. Both reasoner and protege are pretty ancient. I will contact you tomorrow to discuss this issue further.

**Comment by sbello** [comment]
Given historical issues with switching Protege versions, we tend to just stick with what is working

**Comment by sbello** [comment]
So I've now triggered the same error by making a change in the EQ while in the label view. When I switched  to IRI to search for the rogue relations I had this:
![protege_error](https://user-images.githubusercontent.com/10979793/54039837-eafca500-4191-11e9-95e8-993e84eec768.PNG)
@anna-anagnostop please don't make any changes to the EQs while in label view until we figure out what is going on. We should probably look into updating our Protege version, hopefully the newer versions don't have this issue.

**Comment by sbello** [comment]
after - this is used in 4 terms related to diastolic pressure in each case the definition has the phrase
(after some 'cardiac muscle contraction) to capture that this is the pressure between beats when the heart muscle is relaxed
This is in contrast to the systolic terms that use 'exists during'. Which is also a rogue relation

**Comment by matentzn** [comment]
Hey Sue, would you mind trying the installing the latest protege version **in parallel to the one you have** (so you can continue using the old one if problems occur) and repeat the steps you described above and see whether the problem persists? @rsgoncalves is a member of the Protege team, and he will relay the issue to the development team. If you can make it a proper recipe: (1) I click this, (2) add that (3) change this (4) change that (similar to what you did, with a spice more detail), that would be great and help Rafael with debugging! Thaanks a lot!

**Comment by rsgoncalves** [comment]
I couldn't quite pin down the problem looking at the above posts, so as @matentzn said, it would really help to have detailed steps to replicate the issue. If possible, please try those steps in the latest Protege 5.5 as there are substantial changes. Thanks :) I'll stay tuned

**Comment by sbello** [comment]
I'll work on getting the 5.5 version on my machine next week. I have some other things I need to get done today.

**Comment by sbello** [comment]
@rsgoncalves @matentzn I installed 5.5 and tried to recreate the bug and I am still seeing the same bug in 5.5
Steps:
1. open mp-edit.owl
2. select term, I picked abnormal neuron physiology (MP:0004811) at random
3. in label view double click on the Equivalent to field to open the EQ
4. click OK to close the EQ
5. switch to IRI view and see the mess
Pictures of the EQ before
label: 
![EQ_pre_label](https://user-images.githubusercontent.com/10979793/54307613-77e49b80-45a2-11e9-8e4b-f6409898646a.PNG)
IRI:
![EQ_pre_IRI](https://user-images.githubusercontent.com/10979793/54307643-829f3080-45a2-11e9-95f4-34a138d66b8b.PNG)

Pictures of the EQ after
label (note no apparent change)
![EQ_post_label](https://user-images.githubusercontent.com/10979793/54307671-99de1e00-45a2-11e9-8c1d-292743ce8018.PNG)
IRI (note all the plain text in the EQ now)
![EQ_post_IRI](https://user-images.githubusercontent.com/10979793/54307696-a9f5fd80-45a2-11e9-8e22-25a6f065b37e.PNG)

**Comment by sbello** [comment]
I did not run the reasoner
I did not make any changes to the EQ just opened and closed using OK.

**Comment by sbello** [comment]
Same thing happens if I click the edit button to open the EQ
Same thing happens if I click cancel to close
Same thing happens if I click the red X to close
Another example where terms appear to get broken down into component parts
term is abnormal blood-brain barrier function
pre-IRI
![EQ_pre_bloodbrain](https://user-images.githubusercontent.com/10979793/54309529-b54b2800-45a6-11e9-9814-c00b9d0ae71c.PNG)
post IRI
![EQ_post_bloodbrain](https://user-images.githubusercontent.com/10979793/54309542-bd0acc80-45a6-11e9-9d88-f517e1d432b5.PNG)

**Comment by sbello** [comment]
@dosumis Suggested this might be a rendering issue and not a real corruption. To test I did the steps above, saved the file, then reopened the file and checked the EQ. Turns out David is correct, when the file is reopened the IRI display is back to normal. So weird rendering issue but no permanent change to the file.

**Comment by matentzn** [comment]
Wow, seems the bug was fixed for Protege 5.6 (next version). @rsgoncalves this still leaves open the question of how Protege label rendering disambiguates duplicate labels. We have to create annoying workarounds such as [this one](https://github.com/obophenotype/human-phenotype-ontology/wiki/Protege-and-label-based-editing-(PATO:quality-vs-BFO:quality)) to deal with it. Now I see you have this cool new feature with adding the short form in brackets (loving it!). Can you see a user friendly way to add that to the class expression editor to avoid the duplicate label problem?

**Comment by rsgoncalves** [comment]
Thanks Nico. We'll have a look at your post and create an issue for it.

**Comment by matthewhorridge** [comment]
> Can you see a user friendly way to add that to the class expression editor to avoid the duplicate label problem?

Will look into this, but my initial thought is that it sounds doable.  Might require some not insignificant changes to the OWL API parser though.

**Comment by matthewhorridge** [comment]
Just a heads up for this:

> this still leaves open the question of how Protege label rendering disambiguates duplicate labels. We have to create annoying workarounds such as this one to deal with it. 

5.6.0 will select a term that is defined in the active ontology over one that isn't, and then select a term that is most commonly used in the active ontology over one that isn't.  The upshot of this, for HP, is that typing 'quality' will bind to the PATO term rather than the BFO term.  See screenshots here:

https://github.com/protegeproject/protege/issues/869#issuecomment-479968839

**Comment by matentzn** [comment]
Thanks @matthewhorridge! That sounds great!

**Comment by sbello** [comment]
superseded by #3753
